sstables/sstable_directory.hh

```diff
@@ -189,6 +190,14 @@ private:
     // Retrieves sstables::foreign_sstable_open_info for a particular SSTable.
     future<foreign_sstable_open_info> get_open_info_for_this_sstable(const sstables::entry_descriptor& desc) const;
 
+    sstable_directory(sstables_manager& manager,
+          schema_ptr schema,
+          std::unique_ptr<dht::sharder> sharder_ptr,
+          const dht::sharder* sharder,
```

Comment:
Why do we have two sharder pointers?  Who will pass the unique_ptr version and who will pass the plain pointer version? Can we unify them?

Reply:
This is private interface, not pretty I know, they're mutually exclusive, will try to unify them using std::variant<...>, so that's made explicit in the interface.